### ISSUE TITLE
Index variation in a coupler branch

### ISSUE THREAD

**Issue by solenec08** [NanoComp/meep#1637]
Hello,

I am a new user of Meep. I followed the tutorial on GDSII Import and based my script on examples/coupler.py.

I would like to create a gradual index variation within the two branches of my coupler. That is why I tried to put a function as first argument of mp.get_GDSII_prisms. However, this doesn't seem to work and I have the error "OverflowError: in method 'get_GDSII_prisms', argument 3 of type 'int'"

I thought that any function that accepts a mp.Medium instance could also take a user-defined function. How can I create such a variation of the index ?

Thank you in advance

**Comment by stevengj** [comment]
This should work. Possibly you defined the material function incorrectly?   Try using:
```py
def test_material_function(p):
    return mp.Medium(index=2)
```
for example. 

(Or maybe you have some unrelated bug in how you are calling it; it's impossible to say for sure without seeing example code illustrating the problem.)

**Comment by solenec08** [comment]
Thank you for your answer. I tried to add test_material_function directly in the script from the GDSII-import tutorial, but I still have the same issue...

**Comment by stevengj** [comment]
Looks like there were a couple of problems with using material functions with prisms.  They should be fixed by #1652.
